grafana/grafana
# Query History: Implement RemoteStorage methods: settings

**What this PR does / why we need it**:

This PR adds methods to update and load user preferences for query history. The settings can be changes in the Query History settings tab:

<img width="508" alt="Screen Shot 2022-05-20 at 13 32 12" src="https://user-images.githubusercontent.com/745532/169520454-cf74c99c-1005-4d96-b40a-fc80937c9d52.png">

When query history is disabled it shows the old settings panel with more options. 

**Which issue(s) this PR fixes**:



Fixes #48408

**Special notes for your reviewer**:

How to test it?

Make sure query history is enabled in `custom.ini`:

```
[query_history]
enabled = true
```

Change settings to make starred tab default and check if it's selected when the container is opened (also after refreshing the page).

There's one caveat: There's a bug when settings are updated from User Preferences section. It overrides all existing preferences. `queryHistory` settings are passed to keep them intact but they are not being saved to the database. I've created an issue for it #49322.
